wizeline/falcon/middlewares/bodyParser.py

```diff
@@ -0,0 +1,58 @@
+import json
+from json.decoder import JSONDecodeError
+
+import falcon
+
+class BodyParserMiddleware:
+    def process_resource(self, req, resp, resource, params):
+        if self._is_middleware_enabled(resource) and self._request_method_has_payload(req):
+            if self._is_json_content_type(req):
+                try:
+                    req.text = self._get_payload(req)
+
+                    if req.text.strip() != '':
+                        req.json = json.loads(req.text)
+                    else:
+                        req.json = {}
+
+                except JSONDecodeError:
+                    raise falcon.HTTPInternalServerError()
+            elif self._is_urlencoded_content_type(req):
+                req.json = req.params
+            else:
+                raise falcon.HTTPUnsupportedMediaType()
+
+    def _is_middleware_enabled(self, resource):
+        return not hasattr(resource, 'disable_json_middleware') \
```

Comment:
Why do we need to have a property for disabling the middleware?

Reply:
This is needed if some resource will parse the content differently (for example, a request containing a file). I would suggest changing the code to
```py
return not getattr(resource, 'disable_json_middleware', False)
```
so it is easier to read.